Apply architecture feedback (#3447)
viamrobotics/docs: docs/architecture/_index.md

If you have a more complex situation with multiple computers and associated hardware working together, you have two options for organization:

- One complex {{< glossary_tooltip term_id="machine" text="machine" >}} consisting of multiple parts, working together.
- Multiple individual machines (each made up of one or more parts), linked by a {{< glossary_tooltip term_id="remote-part" text="remote" >}} connection.

These two options are very similar: in both cases, the parts communicate with each other securely and directly using gRPC/{{< glossary_tooltip term_id="webrtc" text="WebRTC" >}}.
Any given part can be a remote part of multiple machines, whereas a part can only be a sub-part of one machine.
In other words, remote connections allow sharing of resources across multiple machines, whereas main parts and sub-parts are a way to hierarchically organize one machine.

> Connecting parts (either as main part and sub-part, or as part and remote part) means that you can write SDK code that establishes a connection with one part and controls them all in a coordinated way.

Reviewer: would you have to connect to the main part though?

Author: Yes, but since with remotes you don't talk about "main part" in the same way I left it vague. However, maybe can gloss over and make this clearer:
```suggestion
Connecting parts (either as main part and sub-part, or as part and remote part) means that you can write SDK code that establishes a connection with the main part and controls all parts in a coordinated way.
```
